DisplayBundle/BBcode/InternalLinkDefinition.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+namespace OpenOrchestra\DisplayBundle\BBcode;
+
+use OpenOrchestra\BBcodeBundle\Definition\BBcodeDefinition;
+use OpenOrchestra\BBcodeBundle\ElementNode\BBcodeElementNodeInterface;
+use OpenOrchestra\BBcodeBundle\ElementNode\BBcodeElementNode;
+use Symfony\Component\Templating\EngineInterface;
+use OpenOrchestra\DisplayBundle\Manager\NodeManager;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+
+
+use Symfony\Component\HttpFoundation\ParameterBag;
+use Symfony\Component\Routing\RequestContext;
+
+/**
+ * Class InternalLinkDefinition
+ */
+class InternalLinkDefinition extends BBcodeDefinition
+{
+
+    protected $urlGenerator;
+    protected $nodeManager;
+    protected $templating;
+
+    /**
+     * @param UrlGeneratorInterface $urlGenerator
+     * @param NodeManager           $nodeManager
+     * @param EngineInterface       $templating
+     */
+    public function __construct(
+        UrlGeneratorInterface $urlGenerator,
+        NodeManager $nodeManager,
+        EngineInterface $templating
+    ){
+        parent::__construct('link', '');
+
+        $this->urlGenerator = $urlGenerator;
+        $this->nodeManager = $nodeManager;
+        $this->templating = $templating;
+        $this->useOption = true;
+    }
+
+    /**
+     * Returns this node as HTML
+     *
+     * @return string
+     */
+    public function getHtml(BBcodeElementNode $el)
+    {
+        return $this->generateHtml($el);
+    }
+
+    /**
+     * Returns this node as HTML, in a preview context
+     *
+     * @return string
+     */
+    public function getPreviewHtml(BBcodeElementNodeInterface $el)
+    {
+        return $this->generateHtml($el, true);
+    }
+
+    /**
+     * @param BBcodeElementNodeInterface $el
+     * @param bool                       $preview
+     */
+    protected function generateHtml(BBcodeElementNodeInterface $el, $preview = false)
+    {
+        $children = $el->getChildren();
+        $option = $el->getAttribute();
+        if ($preview) {
```

Comment:
I don't think so. generateHtml with preview = true is used in backoffice from getPreviewHtml method (and is not seeing front routers) and preview = false in front with routers from getHtml @nicolasThal